src/Surfnet/ServiceProviderDashboard/Application/Metadata/JsonGeneratorStrategy.php

```diff
@@ -50,9 +50,16 @@ public function addStrategy($identifier, GeneratorInterface $generator): void
     /**
      * @throws JsonGeneratorStrategyNotFoundException
      */
-    public function generateForNewEntity(ManageEntity $entity, string $workflowState): array
-    {
-        return $this->getStrategy($entity->getProtocol()->getProtocol())->generateForNewEntity($entity, $workflowState);
+    public function generateForNewEntity(
```

Comment:
Not the cause of this change and probably out of scope, but it looks weird to me whe we call generateForNewEntity, that it returns a strategy? Not sure what a strategy is in this context, but the naming could be a bit more clarifying. Like generateStrategyForNewEntity? And return a Startegy object?

Reply:
It first uses the strategy pattern to get the correct JSON generator for the applicable Manage Entity protocol, and then generates' n returns the JSON